index.js

```diff
@@ -219,41 +164,43 @@
       };
     };
 
-    switch (typeOf(x)) {
```

Comment:
It was part of why quest to reduce the number of helper functions. It was only used in two places so I decided it wasn't pulling its weight. `toString.call(x)` has the advantage of being more explicit too, once one is aware that `toString` is a reference to `Object.prototype.toString`.

I'm happy to reinstate `typeOf` if you think it improves readability.